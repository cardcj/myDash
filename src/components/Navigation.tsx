import * as React from 'react';
import styled from '@emotion/styled';

const NavButton = styled('li')`
	display:inline-block;
	text-align: center
	padding: 10px;
	margin: 10px;
	font-family: 'Acme', sans-serif;
	font-size: 20px;
	list-style: none;
	cursor: pointer;
	&:after {
		display: block;
		content: '';
		border-bottom: 2px aquamarine solid;
		transform: scaleX(0);
		transition: transform 200ms ease-in-out;
	}
	&:hover:after {
		transform: scaleX(1);
	}
`

const NavBar = styled('ul')`
	width: 80%;
	margin: 0 auto;
	text-align: center;
	border-bottom: thin lightgrey solid;
`

const Navigation = () => {
	return (
		<NavBar>
			<NavButton>To-Do</NavButton>
			<NavButton>Journal</NavButton>
			<NavButton>Finances</NavButton>
			<NavButton>Timetable</NavButton>
		</NavBar>
	);
}

export default Navigation;