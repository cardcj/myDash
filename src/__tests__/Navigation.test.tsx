import React from 'React';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';
import Navigation from '../components/Navigation';

it('renders without crashing', () => {
    const div = document.createElement('div');
    render(<Navigation />, div);
});

it('checks against last snapshot', () => {
    const tree = renderer.create(<Navigation />).toJSON();
    expect(tree).toMatchSnapshot();
});