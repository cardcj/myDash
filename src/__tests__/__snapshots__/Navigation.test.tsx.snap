// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks against last snapshot 1`] = `
<nav
  className="css-xy7zbc"
  role="navigation"
>
  <li
    className="css-1z0o9ve"
  >
    To-Do
  </li>
  <li
    className="css-1z0o9ve"
  >
    Journal
  </li>
  <li
    className="css-1z0o9ve"
  >
    Finances
  </li>
  <li
    className="css-1z0o9ve"
  >
    Timetable
  </li>
</nav>
`;
